clang/lib/CIR/Dialect/Transforms/TargetLowering/Targets/SPIR.cpp

```diff
@@ -37,10 +38,27 @@ class SPIRVABIInfo : public ABIInfo {
   }
 };
 
+//===----------------------------------------------------------------------===//
+// SPIR-V other target-specific information
+//===----------------------------------------------------------------------===//
+
+static AddressSpaceAttr::MapTy spirvAddrSpaceMap = {
+    0, // None
+    0, // offload_private
+    3, // offload_local
+    1, // offload_global
+    2, // offload_constant
+    4, // offload_generic
+};
```

Comment:
What happens if a target define an address space map that has fewer, more or none elements? Why this need to be a static global? Seems like a constant global would be enough.

Reply:
> In other words, all maps for different targets should encode all cases that is always sync with `AddressSpaceAttr`.

My question is a more practical one: if by mistake the array is incomplete, do we get a compiler error or we gonna silently miscompile? A lookup tables are cool, but if you only use it once it might not be worth the potential mistakes other could make.